src/astro/sql/operators/transform.py

```diff
@@ -0,0 +1,397 @@
+
+    :param python_callable: Function to decorate
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. List/Tuples will unroll to xcom values
+        with index as key. Dict will unroll to xcom values with keys as XCom keys.
+        Defaults to False.
+    :type multiple_outputs: bool
+    """
+
+    return task_decorator_factory(
+        python_callable=python_callable,
+        multiple_outputs=multiple_outputs,
+        decorated_operator_class=TransformOperator,  # type: ignore
+        **kwargs,
+    )
+
+
+def transform_decorator(
+    python_callable: Optional[Callable] = None,
+    multiple_outputs: Optional[bool] = None,
+    conn_id: str = "",
+    parameters: Optional[Union[Mapping, Iterable]] = None,
+    database: Optional[str] = None,
+    schema: Optional[str] = None,
+    raw_sql: bool = False,
+    handler: Optional[Callable] = None,
+    **kwargs,
+):
+    """
+    :param python_callable: A reference to an object that is callable
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. Dict will unroll to xcom values with keys as keys.
+        Defaults to False.
+    :param conn_id: The reference to a specific Database for input and output table.
+    :param autocommit: if True, each command is automatically committed. (default value: False)
+    :param parameters: The parameters to render the SQL query with.
+    :param database: name of database which overwrite defined one in connection
+    """
+    return _transform_task(
+        python_callable=python_callable,
+        multiple_outputs=multiple_outputs,
+        conn_id=conn_id,
+        parameters=parameters,
+        database=database,
+        schema=schema,
+        raw_sql=raw_sql,
+        handler=handler,
+        **kwargs,
+    )
+
+
+def load_op_arg_dataframes_into_sql(
+    conn_id: str, op_args: Tuple, target_table: Table
+) -> Tuple:
+    """
+    Identifies dataframes in op_args and loads them to the table
+
+    :param conn_id:
+    :param op_args:
+    :param target_table:
+    :return:
+    """
+    final_args = []
+    database = create_database(conn_id=conn_id)
+    for arg in op_args:
+        if isinstance(arg, pd.DataFrame):
+            database.load_pandas_dataframe_to_table(
+                source_dataframe=arg, target_table=target_table
+            )
+            final_args.append(target_table)
+        elif isinstance(arg, Table):
+            arg = database.populate_table_metadata(arg)
+            final_args.append(arg)
+        else:
+            final_args.append(arg)
+    return tuple(final_args)
+
+
+def load_op_kwarg_dataframes_into_sql(
+    conn_id: str, op_kwargs: Dict, target_table: Table
+) -> Dict:
+    """
+    Identifies dataframes in op_kwargs and loads them to the table
+
+    :param conn_id:
+    :param op_kwargs:
+    :param target_table:
+    :return:
```

Comment:
?

Reply:
I'm not sure what you're asking here

Reply:
i.e. all these don't have docstrings / descriptions